src/java/org/apache/cassandra/io/sstable/Descriptor.java

```diff
@@ -62,6 +65,10 @@
     public final SSTableFormat.Type formatType;
     private final int hashCode;
 
+    private final String baseFileURI;
+    private final String filenamePart;
+    private final ConcurrentMap<Component, File> filePaths;
```

Comment:
why do we need this cache?

Reply:
reduce `File` creation. it was added in DB-2355

Reply:
I'm sorry but I'm -1 on this change. I does not look like it is needed in the context of this ticket and it needlessly complicates the code by turning a pretty nice immutable object into mutable. 

If you insist on this change, please add a microbenchmark which can somehow certify this change

Reply:
removed

Reply:
thank you